### ISSUE TITLE
Zaproxy not generating HTML report in Azure Pipeline

### ISSUE THREAD

**Issue by duttongeorge13** [zaproxy/zaproxy#8634]
### Describe the bug

I am trying to generate a Zaproxy HTML report for my project in an Azure Pipeline, but the report isn't being created.

This is the docker command I am using to run Zaproxy:

 docker run --rm \
        --net zapnet \
        -v  $(pwd):/zap/wrk/:rw \
        -t  zaproxy/zap-stable zap-full-scan.py \
        -t http://$(ip -f inet -o addr show docker0 | awk '{print $4}' | cut -d '/' -f 1):8080 \
        -r report.html \
        -reportDir '$(System.DefaultWorkingDirectory)'

I have done this exact same command with the XML, JSON and MD reports and it generates them in the working directory without issue.

There are no errors or warnings in the console and it's definitely running Zaproxy without issue as there are lines in the console telling me I have passed all the criteria, same as with the XML, JSON and MD reports.

![Zaproxy](https://github.com/user-attachments/assets/64f53d20-eda6-4e40-b022-622723b0af0a)

I have used Get-ChildItem "$(System.DefaultWorkingDirectory)" to check the directory and the html report is not there. Again, running this with the XML, JSON and MD reports shows the generated report in the directory; it's just the html report that isn't.

### Steps to reproduce the behavior

1. Build project docker image.
2. Use docker run --net zapnet to create a network.
3. Run Zapnet to generate a html report using the following command: docker run --rm \
        --net zapnet \
        -v  $(pwd):/zap/wrk/:rw \
        -t  zaproxy/zap-stable zap-full-scan.py \
        -t http://$(ip -f inet -o addr show docker0 | awk '{print $4}' | cut -d '/' -f 1):8080 \
        -r report.html \
        -reportDir '$(System.DefaultWorkingDirectory)'
4. Html report isn't generated and publishing the report as an artifact fails.

### Expected behavior

Html report is generated in the project working directory.

### Software versions

zaproxy/zap-stable:latest docker image. I believe this is 2.15.0 as of writing.

### Would you like to help fix this issue?

- [ ] Yes

**Comment by duttongeorge13** [comment]
> Where did you see the `-reportDir` ? That does not exist in the `zap-full-scan.py`. Did you check the `pwd`? That's where the reports would be saved given the arguments provided.

I can't recall. I wrote this  a few months previously and was recently tasked with outputting the results to a html report instead of an md report.

I removed the -reportDir parameter and the html report is now being output to the working directory. I'm very confused though because the XML, JSON and MD reports didn't work previously until I added the reportDir parameter, but the html report only works without it.
